src/caseless.rs

```diff
@@ -0,0 +1,134 @@
+//! This module contains a caseless filesystem.
+
+use std::ffi::OsString;
+use std::io;
+use std::path::{Component, Path, PathBuf};
+
+use crate::index::normalize_path;
+use crate::prelude::*;
+use crate::store::Entries;
+
+/// Caseless filesystem wrapping an inner filesystem.
```

Comment:
Could you mention that "caseless" means "case insensitive paths"?

Reply:
Ok.